system/include/simd128.h

```diff
@@ -0,0 +1,747 @@
+/*
+WebAssembly SIMD128 Intrinsics
+*/
+
+#include <stdint.h>
```

Comment:
Is it bad to have this header file include other header files? We could use the same types as the builtins (vector of char, short, int, etc.) instead of using the C99 stdint types. This would slightly less nice from a typing point of view, but would be simpler and mostly invisible to users.

Reply:
It is probably OK to include other headers, I see a few system headers in Clang's sources include `stdint.h`, and quite a few also include each other.  On the other hand, converting between external and internal types might not be necessary as it would work with out it and it is generally a good idea to reduce amount of C-style casting :)

Reply:
Unfortunately I was getting type errors when trying to use the user-facing integer vector types with the builtin functions, so those casts are necessary. They have no runtime impact, though.

Reply:
Yes, that makes sense. What I was thinking, is that user-facing type would be the same as internal, so casts would not be needed. I don't feel strongly either way though.